controllers/internalrequest/adapter.go

```diff
@@ -18,34 +18,134 @@ package internalrequest
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+	"knative.dev/pkg/apis"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
 )
 
 // Adapter holds the objects needed to reconcile a Release.
 type Adapter struct {
 	client          client.Client
-	context         context.Context
+	ctx             context.Context
 	internalClient  client.Client
 	internalRequest *v1alpha1.InternalRequest
+	loader          loader.ObjectLoader
 	logger          logr.Logger
 }
 
 // NewAdapter creates and returns an Adapter instance.
-func NewAdapter(internalRequest *v1alpha1.InternalRequest, client, internalClient client.Client, context context.Context, logger logr.Logger) *Adapter {
+func NewAdapter(ctx context.Context, client, internalClient client.Client, internalRequest *v1alpha1.InternalRequest, loader loader.ObjectLoader, logger logr.Logger) *Adapter {
 	return &Adapter{
 		client:          client,
-		context:         context,
+		ctx:             ctx,
 		internalRequest: internalRequest,
 		internalClient:  internalClient,
+		loader:          loader,
 		logger:          logger,
 	}
 }
 
-func (a *Adapter) EnsureReconcileOperationIsLogged() (reconciler.OperationResult, error) {
-	a.logger.Info("InternalRequest successfully watched")
+// EnsureRequestIsHandled is an operation that will ensure that the InternalRequest is handled by creating a new
+// PipelineRun for the Pipeline referenced in the Request field.
+func (a *Adapter) EnsureRequestIsHandled() (reconciler.OperationResult, error) {
+	pipelineRun, err := a.loader.GetInternalRequestPipelineRun(a.ctx, a.internalClient, a.internalRequest)
+	if err != nil && !errors.IsNotFound(err) {
+		return reconciler.RequeueWithError(err)
+	}
+
+	if pipelineRun == nil || !a.internalRequest.HasStarted() {
+		if pipelineRun == nil {
+			pipelineRun = tekton.NewPipelineRun(a.internalRequest)
+
+			err = libhandler.SetOwnerAnnotations(a.internalRequest, pipelineRun)
+			if err != nil {
+				return reconciler.RequeueWithError(err)
+			}
+
+			err = a.internalClient.Create(a.ctx, pipelineRun)
+			if err != nil {
+				return reconciler.RequeueWithError(err)
+			}
+
+			a.logger.Info("Created PipelineRun to handle request",
+				"PipelineRun.Name", pipelineRun.Name, "PipelineRun.Namespace", pipelineRun.Namespace)
+		}
+
+		return reconciler.RequeueOnErrorOrContinue(a.registerInternalRequestStatus(pipelineRun))
+	}
 
 	return reconciler.ContinueProcessing()
 }
+
+// EnsureStatusIsTracked is an operation that will ensure that the release PipelineRun status is tracked
+// in the InternalRequest being processed.
+func (a *Adapter) EnsureStatusIsTracked() (reconciler.OperationResult, error) {
+	pipelineRun, err := a.loader.GetInternalRequestPipelineRun(a.ctx, a.internalClient, a.internalRequest)
+	if err != nil && !errors.IsNotFound(err) {
+		return reconciler.RequeueWithError(err)
+	}
+
+	if pipelineRun != nil {
+		klog.Infof("%+v", pipelineRun.Status.Conditions)
```

Comment:
did you mean to leave this here or is just part of your testing? If it is being left there, I'd hope for some more added to the string so I know we know what we are looking at in the log

Reply:
I missed this.